drivers/wifi/wfx200/wfx200_host.c

```diff
@@ -0,0 +1,730 @@
+		sl_wfx_start_ap_callback((sl_wfx_start_ap_ind_t *) event_payload);
+		break;
+	case SL_WFX_STOP_AP_IND_ID:
+		sl_wfx_stop_ap_callback((sl_wfx_stop_ap_ind_t *) event_payload);
+		break;
+	case SL_WFX_RECEIVED_IND_ID:
+		u.ethernet_frame = (sl_wfx_received_ind_t *) event_payload;
+		if (u.ethernet_frame->body.frame_type == 0) {
+			sl_wfx_host_received_frame_callback(u.ethernet_frame);
+		}
+		break;
+	case SL_WFX_SCAN_RESULT_IND_ID:
+		sl_wfx_scan_result_callback((sl_wfx_scan_result_ind_t *) event_payload);
+		break;
+	case SL_WFX_SCAN_COMPLETE_IND_ID:
+		sl_wfx_scan_complete_callback((sl_wfx_scan_complete_ind_t *) event_payload);
+		break;
+	case SL_WFX_AP_CLIENT_CONNECTED_IND_ID:
+		sl_wfx_ap_client_connected_callback((sl_wfx_ap_client_connected_ind_t *) event_payload);
+		break;
+	case SL_WFX_AP_CLIENT_REJECTED_IND_ID:
+		sl_wfx_ap_client_rejected_callback((sl_wfx_ap_client_rejected_ind_t *) event_payload);
+		break;
+	case SL_WFX_AP_CLIENT_DISCONNECTED_IND_ID:
+		sl_wfx_ap_client_disconnected_callback(
+			(sl_wfx_ap_client_disconnected_ind_t *) event_payload);
+		break;
+	case SL_WFX_EXT_AUTH_IND_ID:
+		sl_wfx_ext_auth_callback((sl_wfx_ext_auth_ind_t *) event_payload);
+		break;
+	case SL_WFX_GENERIC_IND_ID:
+		break;
+	case SL_WFX_EXCEPTION_IND_ID:
+		u.firmware_exception = (sl_wfx_exception_ind_t *)event_payload;
+		LOG_ERR("Firmware exception %u", u.firmware_exception->body.reason);
+		LOG_DBG("Header: id(%hhx) info(%hhx)", u.firmware_exception->header.id,
+			u.firmware_exception->header.info);
+		LOG_HEXDUMP_DBG(u.firmware_exception->body.data,
+				u.firmware_exception->header.length - offsetof(sl_wfx_exception_ind_t, body.data),
+				"Body data");
+		break;
+	case SL_WFX_ERROR_IND_ID:
+		u.firmware_error = (sl_wfx_error_ind_t *)event_payload;
+		LOG_ERR("Firmware error %u", u.firmware_error->body.type);
+		LOG_DBG("Header: id(%hhx) info(%hhx)", u.firmware_error->header.id,
+			u.firmware_error->header.info);
+		LOG_HEXDUMP_DBG(u.firmware_error->body.data,
+				u.firmware_error->header.length - offsetof(sl_wfx_error_ind_t, body.data),
+				"Body data");
+		break;
+	}
+
+	k_mutex_lock(&wfx200_0->event_mutex, K_FOREVER);
+	if (wfx200_0->waited_event_id == event_payload->header.id) {
+		if (event_payload->header.length < sizeof(wfx200_0->sl_context.event_payload_buffer)) {
+			/* Post the event into the "queue".
+			 * Not using a queue here. The SiLabs examples use the queue similar to a
+			 * counting semaphore and are not passing the "queue" elements via the "queue"
+			 * but instead with a single buffer. So we can only communicate a single
+			 * element at the same time. The sample code also only adds elements to the "queue",
+			 * which it has been configured to explicit waiting for. A queue is therefore
+			 * not needed.
+			 * */
+			memcpy(wfx200_0->sl_context.event_payload_buffer,
+			       (void *) event_payload,
+			       event_payload->header.length);
+			k_sem_give(&wfx200_0->event_sem);
```

Comment:
@rvoell-gcx 
Semaphore + member is definitely okay for event, but I think there is somewhat more Zephyr-ish way to share such updtes: [Polling API](https://docs.zephyrproject.org/3.0.0/reference/kernel/other/polling.html) 
With [k_poll_signal_raise(&signal, result)](https://docs.zephyrproject.org/3.0.0/reference/kernel/other/polling.html#using-k-poll-signal-raise) it's possible to not just signal readiness, but attach an integer code (result) to it.

So `k_poll_signal_raise `here, in `sl_wfx_host_post_event`, and then `k_poll_signal_check `in `sl_wfx_host_wait_for_confirmation`